diff -Naur Prusa-Firmware-3.9.1/Firmware/Marlin_main.cpp Skelestruder-Firmware-3.9.1/Firmware/Marlin_main.cpp
--- Prusa-Firmware-3.9.1/Firmware/Marlin_main.cpp	2020-09-18 07:59:22.000000000 -0400
+++ Skelestruder-Firmware-3.9.1/Firmware/Marlin_main.cpp	2020-09-21 12:21:44.534497552 -0400
@@ -3246,7 +3246,7 @@
 		st_synchronize();
 
         raise_z_above(MIN_Z_FOR_LOAD, false);
-		current_position[E_AXIS] += 30;
+		current_position[E_AXIS] += 20;
 		plan_buffer_line_curposXYZE(400 / 60); //fast sequence
 		
 		load_filament_final_feed(); //slow sequence
diff -Naur Prusa-Firmware-3.9.1/Firmware/mmu.cpp Skelestruder-Firmware-3.9.1/Firmware/mmu.cpp
--- Prusa-Firmware-3.9.1/Firmware/mmu.cpp	2020-09-18 07:59:22.000000000 -0400
+++ Skelestruder-Firmware-3.9.1/Firmware/mmu.cpp	2020-09-21 12:21:03.853735246 -0400
@@ -631,7 +631,7 @@
 				{ 
 					printf_P(PSTR("Unloading finished 2\n"));
 					disable_e0(); //turn off E-stepper to prevent overheating and alow filament pull-out if necessary
-					delay_keep_alive(MMU_LOAD_TIME_MS);
+					delay_keep_alive(MMU_LOAD_TIME_MS*10);
 					move = MMU_LOAD_MOVE;
 					get_response_print_info(move);
 				}
@@ -1445,9 +1445,9 @@
 //! @retval false Doesn't fit
 static bool can_load()
 {
-    current_position[E_AXIS] += 60;
+    current_position[E_AXIS] += 50;
     plan_buffer_line_curposXYZE(MMU_LOAD_FEEDRATE);
-    current_position[E_AXIS] -= 52;
+    current_position[E_AXIS] -= 42;
     plan_buffer_line_curposXYZE(MMU_LOAD_FEEDRATE);
     st_synchronize();
 
diff -Naur Prusa-Firmware-3.9.1/Firmware/ultralcd.cpp Skelestruder-Firmware-3.9.1/Firmware/ultralcd.cpp
--- Prusa-Firmware-3.9.1/Firmware/ultralcd.cpp	2020-09-21 12:39:35.512977462 -0400
+++ Skelestruder-Firmware-3.9.1/Firmware/ultralcd.cpp	2020-09-21 12:44:04.156436989 -0400
@@ -2561,13 +2561,14 @@
     else
     {
         MENU_ITEM_SUBMENU_P(PSTR("PLA  -  " STRINGIFY(PLA_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(PLA_PREHEAT_HPB_TEMP)),mFilamentItem_PLA);
-        MENU_ITEM_SUBMENU_P(PSTR("PET -  " STRINGIFY(PET_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(PET_PREHEAT_HPB_TEMP)),mFilamentItem_PET);
+        MENU_ITEM_SUBMENU_P(PSTR("PETG -  " STRINGIFY(PET_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(PET_PREHEAT_HPB_TEMP)),mFilamentItem_PET);
         MENU_ITEM_SUBMENU_P(PSTR("ASA  -  " STRINGIFY(ASA_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(ASA_PREHEAT_HPB_TEMP)),mFilamentItem_ASA);
         MENU_ITEM_SUBMENU_P(PSTR("PC   -  " STRINGIFY(PC_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(PC_PREHEAT_HPB_TEMP)),mFilamentItem_PC);
         MENU_ITEM_SUBMENU_P(PSTR("ABS  -  " STRINGIFY(ABS_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(ABS_PREHEAT_HPB_TEMP)),mFilamentItem_ABS);
         MENU_ITEM_SUBMENU_P(PSTR("HIPS -  " STRINGIFY(HIPS_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(HIPS_PREHEAT_HPB_TEMP)),mFilamentItem_HIPS);
         MENU_ITEM_SUBMENU_P(PSTR("PP   -  " STRINGIFY(PP_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(PP_PREHEAT_HPB_TEMP)),mFilamentItem_PP);
         MENU_ITEM_SUBMENU_P(PSTR("FLEX -  " STRINGIFY(FLEX_PREHEAT_HOTEND_TEMP) "/" STRINGIFY(FLEX_PREHEAT_HPB_TEMP)),mFilamentItem_FLEX);
+        MENU_ITEM_FUNCTION_P(PSTR("Nozzle -  " STRINGIFY(FARM_PREHEAT_HOTEND_TEMP) "/0"), mFilamentItem_farm_nozzle);
     }
     if (!eeprom_read_byte((uint8_t*)EEPROM_WIZARD_ACTIVE) && eFilamentAction == FilamentAction::Preheat) MENU_ITEM_FUNCTION_P(_T(MSG_COOLDOWN), lcd_cooldown);
     MENU_END();
@@ -6417,9 +6418,15 @@
 
 	//		extr_unload2();
 
+	current_position[E_AXIS] += 3;
+	plan_buffer_line_curposXYZE(500 / 60);
+	st_synchronize();
 	current_position[E_AXIS] -= 45;
 	plan_buffer_line_curposXYZE(5200 / 60);
 	st_synchronize();
+	current_position[E_AXIS] += 10;
+	plan_buffer_line_curposXYZE(500 / 60);
+	st_synchronize();
 	current_position[E_AXIS] -= 15;
 	plan_buffer_line_curposXYZE(1000 / 60);
 	st_synchronize();
diff -Naur Prusa-Firmware-3.9.1/Firmware/variants/1_75mm_MK3-EINSy10a-E3Dv6full.h Skelestruder-Firmware-3.9.1/Firmware/variants/1_75mm_MK3-EINSy10a-E3Dv6full.h
--- Prusa-Firmware-3.9.1/Firmware/variants/1_75mm_MK3-EINSy10a-E3Dv6full.h	2020-09-18 07:59:22.000000000 -0400
+++ Skelestruder-Firmware-3.9.1/Firmware/variants/1_75mm_MK3-EINSy10a-E3Dv6full.h	2020-09-21 12:15:13.026730381 -0400
@@ -20,7 +20,7 @@
 #define DEVELOPER
 
 // Printer name
-#define CUSTOM_MENDEL_NAME "Prusa i3 MK3"
+#define CUSTOM_MENDEL_NAME "SKELESTRUDER MK3"
 
 // Electronics
 #define MOTHERBOARD BOARD_EINSY_1_0a
@@ -41,7 +41,7 @@
 
 // Steps per unit {X,Y,Z,E}
 //#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,140}
-#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,280}
+#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,490}
 //#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,560}
 
 // Endstop inverting
@@ -208,7 +208,7 @@
 
 #define TMC2130_USTEPS_XY   16        // microstep resolution for XY axes
 #define TMC2130_USTEPS_Z    16        // microstep resolution for Z axis
-#define TMC2130_USTEPS_E    32        // microstep resolution for E axis
+#define TMC2130_USTEPS_E    16        // microstep resolution for E axis
 #define TMC2130_INTPOL_XY   1         // extrapolate 256 for XY axes
 #define TMC2130_INTPOL_Z    1         // extrapolate 256 for Z axis
 #define TMC2130_INTPOL_E    1         // extrapolate 256 for E axis
@@ -263,8 +263,8 @@
 #define TMC2130_SG_THRS_HOME {3, 3, TMC2130_SG_THRS_Z, TMC2130_SG_THRS_E}
 
 //new settings is possible for vsense = 1, running current value > 31 set vsense to zero and shift both currents by 1 bit right (Z axis only)
-#define TMC2130_CURRENTS_H {16, 20, 35, 30}  // default holding currents for all axes
-#define TMC2130_CURRENTS_R {16, 20, 35, 30}  // default running currents for all axes
+#define TMC2130_CURRENTS_H {16, 20, 35, 26}  // default holding currents for all axes
+#define TMC2130_CURRENTS_R {16, 20, 35, 26}  // default running currents for all axes
 #define TMC2130_CURRENTS_R_HOME {8, 10, 20, 18}  // homing running currents for all axes
 // #define TMC2130_UNLOAD_CURRENT_R 12			 // lower current for M600 to protect filament sensor - Unused
 
@@ -491,7 +491,7 @@
  PREHEAT SETTINGS
  *------------------------------------*/
 
-#define FARM_PREHEAT_HOTEND_TEMP 250
+#define FARM_PREHEAT_HOTEND_TEMP 200
 #define FARM_PREHEAT_HPB_TEMP 80
 
 #define PLA_PREHEAT_HOTEND_TEMP 215
diff -Naur Prusa-Firmware-3.9.1/Firmware/variants/1_75mm_MK3S-EINSy10a-E3Dv6full.h Skelestruder-Firmware-3.9.1/Firmware/variants/1_75mm_MK3S-EINSy10a-E3Dv6full.h
--- Prusa-Firmware-3.9.1/Firmware/variants/1_75mm_MK3S-EINSy10a-E3Dv6full.h	2020-09-18 07:59:22.000000000 -0400
+++ Skelestruder-Firmware-3.9.1/Firmware/variants/1_75mm_MK3S-EINSy10a-E3Dv6full.h	2020-09-21 12:15:13.026730381 -0400
@@ -19,7 +19,7 @@
 #define DEVELOPER
 
 // Printer name
-#define CUSTOM_MENDEL_NAME "Prusa i3 MK3S"
+#define CUSTOM_MENDEL_NAME "SKELESTRUDER MK3S"
 
 // Electronics
 #define MOTHERBOARD BOARD_EINSY_1_0a
@@ -43,7 +43,7 @@
 
 // Steps per unit {X,Y,Z,E}
 //#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,140}
-#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,280}
+#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,490}
 //#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,560}
 
 // Endstop inverting
@@ -210,7 +210,7 @@
 
 #define TMC2130_USTEPS_XY   16        // microstep resolution for XY axes
 #define TMC2130_USTEPS_Z    16        // microstep resolution for Z axis
-#define TMC2130_USTEPS_E    32        // microstep resolution for E axis
+#define TMC2130_USTEPS_E    16        // microstep resolution for E axis
 #define TMC2130_INTPOL_XY   1         // extrapolate 256 for XY axes
 #define TMC2130_INTPOL_Z    1         // extrapolate 256 for Z axis
 #define TMC2130_INTPOL_E    1         // extrapolate 256 for E axis
@@ -265,8 +265,8 @@
 #define TMC2130_SG_THRS_HOME {3, 3, TMC2130_SG_THRS_Z, TMC2130_SG_THRS_E}
 
 //new settings is possible for vsense = 1, running current value > 31 set vsense to zero and shift both currents by 1 bit right (Z axis only)
-#define TMC2130_CURRENTS_H {16, 20, 35, 30}  // default holding currents for all axes
-#define TMC2130_CURRENTS_R {16, 20, 35, 30}  // default running currents for all axes
+#define TMC2130_CURRENTS_H {16, 20, 35, 26}  // default holding currents for all axes
+#define TMC2130_CURRENTS_R {16, 20, 35, 26}  // default running currents for all axes
 #define TMC2130_CURRENTS_R_HOME {8, 10, 20, 18}  // homing running currents for all axes
 // #define TMC2130_UNLOAD_CURRENT_R 12			 // lower current for M600 to protect filament sensor - Unused
 
@@ -495,7 +495,7 @@
  PREHEAT SETTINGS
  *------------------------------------*/
 
-#define FARM_PREHEAT_HOTEND_TEMP 250
+#define FARM_PREHEAT_HOTEND_TEMP 200
 #define FARM_PREHEAT_HPB_TEMP 80
 
 #define PLA_PREHEAT_HOTEND_TEMP 215
